docs/querying/sql.md

```diff
@@ -313,46 +313,48 @@ possible for two aggregators in the same SQL query to have different filters.
-|`VAR_SAMP(expr)`|Computes variance sample of `expr`. See [stats extension](../development/extensions-core/stats.md) documentation for additional details.|
-|`VARIANCE(expr)`|Computes variance sample of `expr`. See [stats extension](../development/extensions-core/stats.md) documentation for additional details.|
-|`STDDEV_POP(expr)`|Computes standard deviation population of `expr`. See [stats extension](../development/extensions-core/stats.md) documentation for additional details.|
-|`STDDEV_SAMP(expr)`|Computes standard deviation sample of `expr`. See [stats extension](../development/extensions-core/stats.md) documentation for additional details.|
-|`STDDEV(expr)`|Computes standard deviation sample of `expr`. See [stats extension](../development/extensions-core/stats.md) documentation for additional details.|
-|`EARLIEST(expr)`|Returns the earliest value of `expr`, which must be numeric. If `expr` comes from a relation with a timestamp column (like a Druid datasource) then "earliest" is the value first encountered with the minimum overall timestamp of all values being aggregated. If `expr` does not come from a relation with a timestamp, then it is simply the first value encountered.|
-|`EARLIEST(expr, maxBytesPerString)`|Like `EARLIEST(expr)`, but for strings. The `maxBytesPerString` parameter determines how much aggregation space to allocate per string. Strings longer than this limit will be truncated. This parameter should be set as low as possible, since high values will lead to wasted memory.|
-|`LATEST(expr)`|Returns the latest value of `expr`, which must be numeric. If `expr` comes from a relation with a timestamp column (like a Druid datasource) then "latest" is the value last encountered with the maximum overall timestamp of all values being aggregated. If `expr` does not come from a relation with a timestamp, then it is simply the last value encountered.|
-|`LATEST(expr, maxBytesPerString)`|Like `LATEST(expr)`, but for strings. The `maxBytesPerString` parameter determines how much aggregation space to allocate per string. Strings longer than this limit will be truncated. This parameter should be set as low as possible, since high values will lead to wasted memory.|
-|`ANY_VALUE(expr)`|Returns any value of `expr` including null. `expr` must be numeric. This aggregator can simplify and optimize the performance by returning the first encountered value (including null)|
-|`ANY_VALUE(expr, maxBytesPerString)`|Like `ANY_VALUE(expr)`, but for strings. The `maxBytesPerString` parameter determines how much aggregation space to allocate per string. Strings longer than this limit will be truncated. This parameter should be set as low as possible, since high values will lead to wasted memory.|
-|`GROUPING(expr, expr...)`|Returns a number to indicate which groupBy dimension is included in a row, when using `GROUPING SETS`. Refer to [additional documentation](aggregations.md#grouping-aggregator) on how to infer this number.|
+|Function|Notes|Default|
```

Comment:
I hate to ask, but have you double-checked that for all these aggregators, the three implementations return the same values in the two null-handling modes? There are six cases to check: aggregator, buffer aggregator, vector aggregator vs. sql-compliant, replace-with-default.

Reply:
Or does it not matter because we'll always use one of the impls and never use the other two? (If that's the case, I suggest adding a comment somewhere discussing that, perhaps in TimeseriesQueryQueryToolChest or in the javadocs for the aggregators themselves.)

Reply:
I did eyeball it, but that sounds lame when I say it out loud. I guess the unit tests added will cover heap and vector aggregators for vectorized agg so will confirm at least those match, but the ones which cannot be vectorized only test heap aggs.

I'll add a similar group by query test that uses filtered aggs to match nothing to cover the buffer agg path to make sure everything matches up, which seems pretty useful to have anyway.

Reply:
Added near duplicates of the timeseries tests using group by queries with filtered aggregations that matched nothing to get additional coverage on buffer aggregators and luckily everything matched up.